textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/FuzzyTokenPredicate.java

```diff
@@ -0,0 +1,74 @@
+package edu.uci.ics.textdb.dataflow.common;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IPredicate;
+import edu.uci.ics.textdb.api.storage.IDataStore;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.utils.Utils;
+import edu.uci.ics.textdb.storage.DataReaderPredicate;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.search.Query;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * @author varun bharill, parag saraogi
+ * 
+ * This class builds the query to perform boolean searches in a lucene index. 
+ */
+public class FuzzyTokenPredicate implements IPredicate {
+	
+    private IDataStore dataStore;
+    private String query;
+    private Query luceneQuery;
+    private ArrayList<String> tokens;
+    private List<Attribute> attributeList;
+    private String[] fields;
+    private Analyzer luceneAnalyzer;
+    private double thresholdRatio;
+    private int threshold;
+    private int docCountInSource;
+    private boolean isSpanInformationAdded;
+    
+    public FuzzyTokenPredicate(String query, List<Attribute> attributeList, Analyzer analyzer,IDataStore dataStore, double thresholdRatio, boolean isSpanInformationAdded) throws DataFlowException{
+        try {
+        	this.thresholdRatio = thresholdRatio;
+        	this.dataStore = dataStore;
+        	this.luceneAnalyzer = analyzer;
+        	this.isSpanInformationAdded= isSpanInformationAdded;
```

Comment:
The KeywordSearch operator has too.  It's not required for now, but to be consistent, it will be good to let each operator have it.  @zuozhi you can use a separate PR to add it to your RegexMatcher operator later.